context/src/main/java/io/micronaut/scheduling/processor/ScheduledMethodProcessor.java

```diff
@@ -141,13 +142,16 @@ public void process(BeanDefinition<?> beanDefinition, ExecutableMethod<?, ?> met
             };
 
             String cronExpr = scheduledAnnotation.get(MEMBER_CRON, String.class, null);
+            String zoneIdStr = scheduledAnnotation.get(MEMBER_ZONE_ID, String.class, null);
             String fixedDelay = scheduledAnnotation.get(MEMBER_FIXED_DELAY, String.class).orElse(null);
 
             if (StringUtils.isNotEmpty(cronExpr)) {
                 if (LOG.isDebugEnabled()) {
                     LOG.debug("Scheduling cron task [{}] for method: {}", cronExpr, method);
                 }
-                taskScheduler.schedule(cronExpr, task);
+
+                ScheduledFuture<?> scheduledFuture = taskScheduler.schedule(cronExpr, zoneIdStr, task);
+                scheduledTasks.add(scheduledFuture);
```

Comment:
I neglected to point this out last time.

This line, `scheduledTasks.add(scheduledFuture);` is a notable difference from the original code. I noticed that this is the only `if` block that doesn't add the `scheduledFuture` to the list of `scheduledTasks`.

`scheduledTasks` are used in `close` to cancel in-flight tasks, so I believe this fixes a bug.